In [60]:
# Para cargue de datos desde el directorio local
from google.colab import files
uploaded = files.upload()


Saving Cost_of_Living_Index_by_Country_2024.csv to Cost_of_Living_Index_by_Country_2024.csv


El archivo elegido corresponde a datos del indice de costo de vida por país en el 2024, tomando como referencia la ciudad de NY.

In [61]:
# Validamos el cargue
for filename in uploaded.keys():
    print(f"Archivo subido: {filename}")

Archivo subido: Cost_of_Living_Index_by_Country_2024.csv


In [62]:
# Generamos el dataframe y lo visualizamos
import pandas as pd
medallas_pais = list(uploaded.keys())[0]
df = pd.read_csv(medallas_pais)
print(df.head())

   Rank      Country  Cost of Living Index  Rent Index  \
0     1  Switzerland                 101.1        46.5   
1     2      Bahamas                  85.0        36.7   
2     3      Iceland                  83.0        39.2   
3     4    Singapore                  76.7        67.2   
4     5     Barbados                  76.6        19.0   

   Cost of Living Plus Rent Index  Groceries Index  Restaurant Price Index  \
0                            74.9            109.1                    97.0   
1                            61.8             81.6                    83.3   
2                            62.0             88.4                    86.8   
3                            72.1             74.6                    50.4   
4                            48.9             80.8                    69.4   

   Local Purchasing Power Index  
0                         158.7  
1                          54.6  
2                         120.3  
3                         111.1  
4           

In [69]:
# Validamos el tamaño del dataframe
num_rows, num_cols = df.shape
print(f"Número de filas: {num_rows}")
print(f"Número de columnas: {num_cols}")

Número de filas: 121
Número de columnas: 8


El dataframe contiene información en 8 columnas y 121 filas, es decir datos de 121 países.

In [63]:
# Visualizamos el tipo de dato de cada atributo
print(df.dtypes)

Rank                                int64
Country                            object
Cost of Living Index              float64
Rent Index                        float64
Cost of Living Plus Rent Index    float64
Groceries Index                   float64
Restaurant Price Index            float64
Local Purchasing Power Index      float64
dtype: object


Los datos recolectados son de tipo número excepto el país

In [64]:
df.describe()

Rank  Cost of Living Index  Rent Index  \
count  121.000000            121.000000  121.000000   
mean    61.000000             43.555372   16.052893   
std     35.073732             16.147574   11.412267   
min      1.000000             18.800000    2.400000   
25%     31.000000             30.200000    8.500000   
50%     61.000000             39.500000   12.400000   
75%     91.000000             52.800000   20.100000   
max    121.000000            101.100000   67.200000   

       Cost of Living Plus Rent Index  Groceries Index  \
count                      121.000000       121.000000   
mean                        30.357851        44.228926   
std                         13.263721        17.055109   
min                         11.100000        17.500000   
25%                         19.800000        31.600000   
50%                         27.000000        40.500000   
75%                         37.000000        53.700000   
max                         74.900000       109.100000   

       Restaurant Price Index  Local Purchasing Power Index  
count              121.000000                    121.000000  
mean                36.471074                     65.094215  
std                 18.258110                     39.569094  
min                 12.800000                      2.300000  
25%                 21.600000                     34.800000  
50%                 33.100000                     50.600000  
75%                 47.200000                     99.400000  
max                 97.000000                    182.500000

Enfocandonos en el indice de costo de vida podemos afirmar que se recolectaron 121 datos, que en promedio el costo de vida de estas 121 ciudades es 60% más barato que NY, que el dato mínimo nos informa que hay al menos una ciudad con un costo de vida 81,2% más barato que NY, y el dato máximo nos informa que hay al menos una ciudad con un costo de vida 21% mayor que NY. También nos informa la desviación sts de 16,4 que los datos estan concentrados y cercanos a la media.

In [65]:
df.describe(include = "all")

Rank      Country  Cost of Living Index  Rent Index  \
count   121.000000          121            121.000000  121.000000   
unique         NaN          121                   NaN         NaN   
top            NaN  Switzerland                   NaN         NaN   
freq           NaN            1                   NaN         NaN   
mean     61.000000          NaN             43.555372   16.052893   
std      35.073732          NaN             16.147574   11.412267   
min       1.000000          NaN             18.800000    2.400000   
25%      31.000000          NaN             30.200000    8.500000   
50%      61.000000          NaN             39.500000   12.400000   
75%      91.000000          NaN             52.800000   20.100000   
max     121.000000          NaN            101.100000   67.200000   

        Cost of Living Plus Rent Index  Groceries Index  \
count                       121.000000       121.000000   
unique                             NaN              NaN   
top                                NaN              NaN   
freq                               NaN              NaN   
mean                         30.357851        44.228926   
std                          13.263721        17.055109   
min                          11.100000        17.500000   
25%                          19.800000        31.600000   
50%                          27.000000        40.500000   
75%                          37.000000        53.700000   
max                          74.900000       109.100000   

        Restaurant Price Index  Local Purchasing Power Index  
count               121.000000                    121.000000  
unique                     NaN                           NaN  
top                        NaN                           NaN  
freq                       NaN                           NaN  
mean                 36.471074                     65.094215  
std                  18.258110                     39.569094  
min                  12.800000                      2.300000  
25%                  21.600000                     34.800000  
50%                  33.100000                     50.600000  
75%                  47.200000                     99.400000  
max                  97.000000                    182.500000

In [66]:
import altair as alt
chart = alt.Chart(df).mark_circle().encode(x='Cost of Living Index', y='Rent Index')
chart

/usr/local/lib/python3.10/dist-packages/altair/utils/core.py:384: FutureWarning: the convert_dtype parameter is deprecated and will be removed in a future version.  Do ``ser.astype(object).apply()`` instead if you want ``convert_dtype=False``.
  col = df[col_name].apply(to_list_if_array, convert_dtype=False)


alt.Chart(...)

Este gráfico de dispersión nos afirma que los datos estan mayormente concentrados. Adicional, nos muestra una correlación positiva frente a indice de renta, lo que se traduce que a mayor indice de rente, mayor indice de vida.

In [70]:
missing_data = df.isnull()
missing_data.head()

Rank  Country  Cost of Living Index  Rent Index  \
0  False    False                 False       False   
1  False    False                 False       False   
2  False    False                 False       False   
3  False    False                 False       False   
4  False    False                 False       False   

   Cost of Living Plus Rent Index  Groceries Index  Restaurant Price Index  \
0                           False            False                   False   
1                           False            False                   False   
2                           False            False                   False   
3                           False            False                   False   
4                           False            False                   False   

   Local Purchasing Power Index  
0                         False  
1                         False  
2                         False  
3                         False  
4                         False

Esta consulta nos permite visualizar que al menos en las 4 primeras filas no se tienen datos nulos o perdidos

In [71]:
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("")

Rank
Rank
False    121
Name: count, dtype: int64

Country
Country
False    121
Name: count, dtype: int64

Cost of Living Index
Cost of Living Index
False    121
Name: count, dtype: int64

Rent Index
Rent Index
False    121
Name: count, dtype: int64

Cost of Living Plus Rent Index
Cost of Living Plus Rent Index
False    121
Name: count, dtype: int64

Groceries Index
Groceries Index
False    121
Name: count, dtype: int64

Restaurant Price Index
Restaurant Price Index
False    121
Name: count, dtype: int64

Local Purchasing Power Index
Local Purchasing Power Index
False    121
Name: count, dtype: int64



Esta consulta nos confirma que para ningún tipo de dato, existen datos nulos o perdidos